tests/Tutor.Infrastructure.Tests/TestData/EventStore/QueryBuilder.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Tutor.Infrastructure.Tests.TestData.EventStore.Query;
+
+namespace Tutor.Infrastructure.Tests.TestData.EventStore
+{
+    public static class QueryBuilder
+    {
+        public static IEnumerable<Query> BuildAllPossibleQueries(IEnumerable<DateTime> afterParameters,
+                                                                 IEnumerable<DateTime> beforeParameters,
+                                                                 IEnumerable<ConditionParameter> conditionParameters)
+        {
```

Comment:
The way this method is implemented probably results in way too many test cases, but I couldn't really think of a way to generate queries that results in less test cases, but still covers as many options as possible. The tests still run pretty quickly though, on my end it takes about 4 seconds to run around 1000 queries.

Reply:
I implemented a version that generated less queries in 90ca58eed68402dfb1b08202045c985b32aa58e3.

Reply:
Looking at this closer, I find it pretty challenging to wrap my head around. The whole namespace is full of helper classes to test the QueryInterface. Is there a way to simplify the test design?

Reply:
The goal was to somehow generate a wide variety of test cases for better coverage since there are many ways to use IEventQueryable and many things that can go wrong, but it did end up way too complex and confusing. I refactored it quite a bit in 125b47c6a721df6ecd9e083acde1a46321a70a8f, it's still somewhat complex but I believe it should be more readable now (and it should also be much easier to add any future IEventQueryable methods to these tests).